src/console/Console.cxxtest

```diff
@@ -270,3 +270,37 @@ TEST(ConsoleTest, testAddCommandFlow)
 }
 
 
+static Console::CommandStatus echo_command(FILE *fp, int argc, const char *argv[], void *context)
+{
+    if (argc == 0) {
+        fprintf(fp, "the echo command\n");
+    } else if (argc == 2) {
+        fprintf(fp, "%s\n", argv[2]);
+    } else {
+        return Console::COMMAND_ERROR;
+    }
+
+    return Console::COMMAND_OK;
+}
+
+TEST(ConsoleTest, testQuotedParameters)
+{
+    g_console.add_command("echo", echo_command, nullptr);
+    
+    int s = ConnectSocket("localhost", 12345);
+    
+    EXPECT_EQ(::read(s, buf, 1024), 2);
+    EXPECT_TRUE(!strncmp(buf, "> ", 2));
+    
+    EXPECT_EQ(::write(s, "echo \"It couldn't\"\n",19), 19);
+    usleep(1000);
+    EXPECT_EQ(::read(s, buf, 1024), 14);
+    EXPECT_TRUE(!strncmp(buf, "It couldn't\n> ", 14));
+    
+    EXPECT_EQ(::write(s, "echo '\"Oh no, no, no\", said the man'\n", 37), 37);
+    usleep(1000);
+    EXPECT_EQ(::read(s, buf, 1024), 32);
+    EXPECT_TRUE(!strncmp(buf, "\"Oh no, no, no\", said the man\n> ", 32));
```

Comment:
consider adding another test case with mismatched quote to exercise the error path as well. This will be helpful to keep test coverage high.